go/worker/storage/worker.go

```diff
@@ -95,9 +97,13 @@ func (w *Worker) registerRuntime(commonNode *committeeCommon.Node, checkpointerC
 		}
 	}
 
-	localStorage, err := NewLocalBackend(commonNode.Runtime.DataDir(), id)
-	if err != nil {
-		return fmt.Errorf("can't create local storage backend: %w", err)
+	storageCtor := func(ctx context.Context) (storageApi.LocalBackend, error) {
```

Comment:
Is there a reason why we don't create the local storage backend in `committee/node`? This would avoid passing in this factory function.